aspnet/fundamentals/localization.rst

```diff
@@ -1,8 +1,285 @@
+
+#. ``QueryStringRequestCultureProvider``
+#. ``CookieRequestCultureProvider``
+#. ``AcceptLanguageHeaderRequestCultureProvider``
+
+The default list goes from most specific to least specific. Later in the article I'll show how you can change the order and even add a custom localization provider. If there are no non-null providers, the ``DefaultRequestCulture`` is used.
+    
+QueryStringRequestCultureProvider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In addition the providing the primary mechanism some app will use to set the culture and UI culture, adding a query string to the URL is useful for debugging and testing code. Unless you change the ``RequestCultureProviders`` list, a query string to the will always win as the localization provider. You pass the query string parameters ``culture`` and ``ui-culture``. The following example sets the specific culture (language and region) to Spanish/Mexico:
+
+``http://localhost:5000/?culture=es-MX&ui-culture=es-MX``
+
+If you only pass in one of the two (``culture`` or ``ui-culture``), the query string provider will set both values using the one you passed in. For example, setting just the culture will set both the ``Culture`` and the ``UICulture``:
+
+``http://localhost:5000/?culture=es-MX``
+
+CookieRequestCultureProvider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Production apps will often provide a mechanism to set the culture with the ``ASPNET_CULTURE`` cookie. The sample app provides code to set the cookie for French and English cultures.
+
+The Accept-Language HTTP header
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Accept-Language HTTP header from a browser request is not an infallible way to detect the user's preferred language. This setting indicates what the browser has been set to send or has inherited from the underlying operating system. A production app should include a way for a user to customize their choice of culture.
+
+Use ``RequestLocalizationOptions`` to add or remove localization providers. 
+
+Suppose you want to let your customers store their language and culture in your databases. You could write a provider to look up these values for the user. The following code shows how to add a custom provider:
+
+.. code-block:: c#
+
+    services.Configure<RequestLocalizationOptions>(options =>
+    {
+        var supportedCultures = new[]
+        {
+            new CultureInfo("en-US"),
+            new CultureInfo("fr")
+        };
+
+        options.DefaultRequestCulture = new RequestCulture(culture: "en-US", uiCulture: "en-US");
+        options.SupportedCultures = supportedCultures;
+        options.SupportedUICultures = supportedCultures;
+
+        options.RequestCultureProviders.Insert(0, new CustomRequestCultureProvider(async context =>
+        {
+           // My custom request culture logic
+          return new ProviderCultureResult("en");
+        }));
+    });
+
+Configuring localization
```

Comment:
I think this section should go earlier in the doc so that it's obvious where the various localization services come from.